### ISSUE TITLE
[MDEP-680] dependency:go-offline does not use repositories from parent pom in reactor build

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#1203]
**[Andreas Janning](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=theo)** opened **[MDEP-680](https://issues.apache.org/jira/browse/MDEP-680?redirect=false)** and commented

### Observed behaviour

When running dependency:go-offline on multi-module projects, sub-modules always resolve their dependencies from maven-central, even if they inherit \<repositories> from their parent pom

### Expected behaviour

dependency:go-offline resolves dependencies exactly the same way as they would be resolved in when running mvn install on the project, respecting \<repositories> defined in parent poms.

### Description

When using dependency:go-offline to resolve the dependencies of a reactor build, repositories defined in a parent pom are ignored when resolving the dependencies of a module.

I have attached an example project to illustrate the issue

When running 

```java
mvn org.apache.maven.plugins:maven-dependency-plugin:3.1.2:go-offline -Dmaven.repo.local=./depPluginRepo
```

on the parent project, the dependencies of the "parent" module are correctly resolved from the jcenter repository. But the dependencies of "child" are resolved from maven central. Even worse: When the dependency in question is not found in maven central, the build fails.

 

 

 

---

**Affects:** 3.1.2

**Attachments:**
- [go-offline-repository-reproducer.zip](https://issues.apache.org/jira/secure/attachment/12997443/go-offline-repository-reproducer.zip) (_2.76 kB_)

**Issue Links:**
- [MDEP-743](https://issues.apache.org/jira/browse/MDEP-743) go-offline doesn't resolve dependencies from non-central repos
 (_**"is duplicated by"**_)
- [MDEP-711](https://issues.apache.org/jira/browse/MDEP-711) go-offline fails to download SNAPSHOT from other repository
 (_**"is duplicated by"**_)

**Remote Links:**
- [GitHub Pull Request #46
](https://github.com/apache/maven-dependency-plugin/pull/46)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [02d0b28601e0410952b5b4301375cf4ba564f908](https://gitbox.apache.org/repos/asf?p=maven-dependency-plugin.git;a=commit;h=02d0b28601e0410952b5b4301375cf4ba564f908).

**Comment by jira-importer** [comment]
**[Piotr Zygielo](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pzygielo)** commented

Hello Apache Team,

What are the chances for new release of MDEP?
